lib/improver/metadata/forecast_times.py

```diff
@@ -0,0 +1,257 @@
+        if t_coord.bounds is not None:
+            time_bounds = np.array(
+                [c.bound for c in t_coord.cells()])
+            required_lead_bounds = (
+                time_bounds - forecast_reference_time_points)
+            required_lead_bounds = np.array(
+                [[b.total_seconds() for b in x]
+                 for x in required_lead_bounds]).astype(fr_type)
+        else:
+            required_lead_bounds = None
+        coord_type = iris.coords.AuxCoord
+        if cube.coords("forecast_period"):
+            if isinstance(
+                    cube.coord("forecast_period"), iris.coords.DimCoord):
+                coord_type = iris.coords.DimCoord
+        result_coord = coord_type(
+            required_lead_times,
+            standard_name='forecast_period',
+            bounds=required_lead_bounds,
+            units="seconds")
+        result_coord.convert_units(result_units)
+        if np.any(result_coord.points < 0):
+            msg = ("The values for the time {} and "
+                   "forecast_reference_time {} coordinates from the "
+                   "input cube have produced negative values for the "
+                   "forecast_period. A forecast does not generate "
+                   "values in the past.").format(
+                       cube.coord("time").points,
+                       cube.coord("forecast_reference_time").points)
+            warnings.warn(msg)
+    else:
+        msg = ("The forecast period coordinate is not available within {}."
+               "The time coordinate and forecast_reference_time "
+               "coordinate were also not available for calculating "
+               "the forecast_period.".format(cube))
+        raise CoordinateNotFoundError(msg)
+
+    result_coord.points = result_coord.points.astype(fp_type)
+    if result_coord.bounds is not None:
+        result_coord.bounds = result_coord.bounds.astype(fp_type)
+
+    return result_coord
+
+
+def rebadge_forecasts_as_latest_cycle(cubes, cycletime):
+    """
+    Function to update the forecast_reference_time and forecast_period
+    on a list of input forecasts to match either a given cycletime, or
+    the most recent forecast in the list (proxy for the current cycle).
+
+    Args:
+        cubes (iris.cube.CubeList):
+            Cubes that will have their forecast_reference_time and
+            forecast_period updated.
+        cycletime (str or None):
+            Required forecast reference time in a YYYYMMDDTHHMMZ format
+            e.g. 20171122T0100Z. If None, the latest forecast reference
+            time is used.
+
+    Returns:
+        iris.cube.CubeList:
+            Updated cubes
+    """
+    if cycletime is None and len(cubes) == 1:
+        return cubes
+    cycle_datetime = (find_latest_cycletime(cubes) if cycletime is None
+                      else cycletime_to_datetime(cycletime))
+    return unify_cycletime(cubes, cycle_datetime)
+
+
+def unify_cycletime(cubes, cycletime):
```

Comment:
Renamed from `unify_forecast_reference_time` for consistency.  This was the only place where "cycletime" was not being used to describe this concept.